File: Binary-Search/search-2d/solution.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row2search = None
        for row in range(len(matrix)):
            col = len(matrix[row])-1
            if matrix[row][0] == target or matrix[row][col] == target:
                return True
            if target > matrix[row][0] and target < matrix[row][col]:
                row2search = row
        if row2search == None:
            return False
        for col in range(len(matrix[row2search])):
            if matrix[row2search][col] == target:
                return True
        return False

File: Binary-Search/search-2d/solution.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function (matrix, target) {
  let rowToSearch = null;
  for (let row = 0; row < matrix.length; row++) {
    let col = matrix[row].length - 1;
    if (matrix[row][0] === target || matrix[row][col] === target) {
      return true;
    }
    if (target > matrix[row][0] && target < matrix[row][col]) {
      rowToSearch = row;
    }
  }
  if (rowToSearch == null) return false;
  for (let col = 0; col < matrix[rowToSearch].length; col++) {
    if (matrix[rowToSearch][col] === target) {
      return true;
    }
  }
  return false;
};
